common/components/utils/GroupAPIUtils.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+
+import type {Project} from '../stores/ProjectSearchStore.js';
+import type {LinkInfo} from '../../components/forms/LinkInfo.jsx'
+import type {FileInfo} from '../common/FileInfo.jsx'
+// import {PositionInfo} from "../forms/PositionInfo.jsx";
+
+export type GroupDetailsAPIData = {|
+    id: string,
+    group_name: string,
+    group_location: string,
+    group_description: string,
+    group_short_description: string,
+    // group_projects: $ReadOnlyArray<Project>,
+    // group_links: $ReadOnlyArray<LinkInfo>,
+    // group_files: $ReadOnlyArray<FileInfo>,
+|};
+
+
+export default class GroupAPIUtils {
+    static fetchGroupDetails(id: number, callback: (ProjectDetailsAPIData) => void, errCallback: (APIError) => void): void {
+        fetch(new Request('/api/groups/' + id + '/', {credentials: 'include'}))
+            .then(response => {
+                if(!response.ok) {
+                    throw Error();
+                }
+                return response.json();
+            })
+            .then(projectDetails => callback(projectDetails))
+            // TODO: Get catch to return http status code
+            .catch(response => errCallback && errCallback({
+                errorCode: response.status,
+                errorMessage: JSON.stringify(response)
+            }));
+  }
+
+
+}
+
+// class ProjectAPIUtils {
```

Comment:
How much of this is useful reference code for finishing the GroupAPIUtils? Can we just link to ProjectAPIUtils and say 'use this as a reference' rather than duplicating what could become stale code?